scripts/search-cfgov.coffee

```diff
@@ -0,0 +1,63 @@
+# Description:
+#   Searches cf.gov's crawled HTML. See https://github.com/cfpb/crawl-cfgov
+#
+# Dependencies:
+#   "underscore": "1.8.3"
+#
+# Configuration:
+#   HUBOT_CRAWL_SEARCH_URL - the API endpoint
+#
+# Commands:
+#   hubot search-cfgov <search_string> - Attempts to find the string within cf.gov's HTML
+#
+# Author:
+#   contolini
+#   chosak
+
+gist = require('quick-gist')
+
+module.exports = (robot) ->
+  _  = require("underscore")
+
+
+  robot.respond /(crawl|crawl-search|searchcfgov|search-cfgov) (.*)$/i, (msg) ->
+    postSearch(msg, msg.match[2])
+
+
+  # the main search and reply function
+  postSearch = (msg, query) ->
+
+    msg.send "Searching consumerfinance.gov for #{query}"
+
+    searchUrl = "#{process.env.HUBOT_CRAWL_SEARCH_URL}/?q=#{query}"
+
+    # msg.send "URL is #{searchUrl}"
+    robot.http(searchUrl)
+         .get() (err, res, body) ->
+
+      if err
+        msg.send("Searching consumerfinance.gov, Got error: #{err}")
+        return
+
+      result = null
+      try
+        result = body
+        today = new Date
+        dd = today.getDate()
+        mm = today.getMonth() + 1
+        yyyy = today.getFullYear()
+        today = mm + '/' + dd + '/' + yyyy
+        desc = "Consumerfinance.gov search results for #{query} on #{today} by #{msg.envelope.user.name}"
```

Comment:
Including the current date here feels less useful than including the date of the crawl. This is a good first implementation, but what do you think about modifying the crawler webservice to return JSON with both the matches and the timestamp of the last commit?

Reply:
Brilliant. 👍